test/system/helpers.bash

```diff
@@ -894,5 +894,58 @@ function _podman_commands() {
     awk '/^Available Commands:/{ok=1;next}/^Options:/{ok=0}ok { print $1 }' <<<"$output" | grep .
 }
 
+###############################
+#  _build_health_check_image  #  Builds a container image with a configured health check
+###############################
+#
+# The health check will fail once the /uh-oh file exists.
+#
+# First argument is the desired name of the image
+# Second argument determines whether the /uh-oh file will be removed once the check failed; this way the container can be restarted.
+#
+
+function _build_health_check_image {
+    local imagename="$1"
+    local cleanfile="$2"
+    # Create an image with a healthcheck script; said script will
+    # pass until the file /uh-oh gets created (by us, via exec)
+    cat >${PODMAN_TMPDIR}/healthcheck <<EOF
+#!/bin/sh
+
+if test -e /uh-oh; then
+    echo "Uh-oh on stdout!"
+    echo "Uh-oh on stderr!" >&2
+    if [[ ! -z "$cleanfile" ]]; then
+        # Now remove the file again, so that we can restart the container
+        rm /uh-oh
+    fi
```

Comment:
Suggestion starting on line 907):
```bash
function _build_health_check_image {
    local imagename="$1"

    local cleanfile=""
    if [[ -n "$2" ]]; then
        cleanfile="rm -f /uh-oh"
    fi
    ......
    echo "Uh-oh on stderr!" >&2
    ${cleanfile}
    exit 1
```
Reason: this is a test that makes sense at image-creation time, not image-healthcheck-run-time. This phrasing makes that more obvious to the reader.